### ISSUE TITLE
Maj+click select ignore the first elemmaj-ent

### ISSUE THREAD

**Issue by xxorax** [lzzluca/MultiSelect#2]
If you maintains MAJ then click at one option, then click at an other option (to select multiple option), then release MAJ and click to an other option, you can see that your first option is not selected anymore.

**Comment by lzzluca** [comment]
Hi,
sorry but not sure I have got the problem: what do you mean by MAJ?
Thanks for the feedback!

**Comment by xxorax** [comment]
Aah sorry, by MAJ in mean SHIFT :)

**Comment by lzzluca** [comment]
Ok! I am going to check this very soon!

**Comment by lzzluca** [comment]
If it is urgent, you can fix the problem replacing `_attachSelectionAtOnChange` with this:
 
```
      _attachSelectionAtOnChange: function() {
        this.change(function(e) {
          var list = $(this).find(":selected"),
              selMethods = $.data(this, "MultiSelect_selection"),
              // the onchange doesn't listen to the keyboard, so this is a
              // workaround: if many elements were selected at once, then I
              // assume the shift key was pressed
              isShiftKeyPressed = list.length > 1,
              $clone = $.data(this, "MultiSelect_clone");
          for (var i = 0, $option, $option_clone; list[i]; i++) {
            $option = $(list[i]);

            // fix for issue #2
            if( isShiftKeyPressed && (this.selectedIndex === $option.get(0).index) ) {
              continue;
            }

            selMethods.toggle( $option.get(0) );
            $option.attr("selected", false);

            // if the clone select is defined, the selection is update on it too
            if( $clone ) {
              var $option_clone = $clone.children()[ $option.prop("index") ];
              $option_clone = $( $option_clone );
              // TODO check "selected" should be equals "selected" or true?
              $option_clone.attr("selected") === "selected" ?
                $option_clone.removeAttr("selected") : $option_clone.attr("selected", "selected");
            }
          }
        });
      },
```
I would like to write a cleaner solution, so not sure I am going for this solution but I have quickly tested it and seems to work!

**Comment by lzzluca** [comment]
PS this is the diff, in case can be helpful:
```
diff --git a/src/multiselect.js b/src/multiselect.js
index 03ddf04..a965cd6 100644
--- a/src/multiselect.js
+++ b/src/multiselect.js
@@ -54,12 +54,22 @@
     var methods = {

       _attachSelectionAtOnChange: function() {
-        this.change(function() {
+        this.change(function(e) {
           var list = $(this).find(":selected"),
               selMethods = $.data(this, "MultiSelect_selection"),
+              // the onchange doesn't listen to the keyboard, so this is a
+              // workaround: if many elements were selected at once, then I
+              // assume the shift key was pressed
+              isShiftKeyPressed = list.length > 1,
               $clone = $.data(this, "MultiSelect_clone");
           for (var i = 0, $option, $option_clone; list[i]; i++) {
             $option = $(list[i]);
+
+            // fix for issue #2
+            if( isShiftKeyPressed && (this.selectedIndex === $option.get(0).index) ) {
+              continue;
+            }
+
             selMethods.toggle( $option.get(0) );
             $option.attr("selected", false);

@@ -68,7 +78,8 @@
               var $option_clone = $clone.children()[ $option.prop("index") ];
               $option_clone = $( $option_clone );
               // TODO check "selected" should be equals "selected" or true?
-              $option_clone.attr("selected") === "selected" ? $option_clone.removeAttr("selected") : $option_clone.attr("selected", "selected");
+              $option_clone.attr("selected") === "selected" ?
+                $option_clone.removeAttr("selected") : $option_clone.attr("selected", "selected");
             }
           }
         });
```

**Comment by lzzluca** [comment]
Now the fix is on master:
https://github.com/lzzluca/MultiSelect/commit/89ff19bb0a236440914f2dbb11a15876992cbea3
